src/mavlink-router/dedup.cpp

```diff
@@ -0,0 +1,92 @@
+/*
+ * This file is part of the MAVLink Router project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+#include "dedup.h"
+
+#include <chrono>
+#include <string>
+#include <queue>
+#include <unordered_set>
+
+class DedupImpl {
+    using hash_t = uint64_t;
+    using time_t = uint32_t;
+
+public:
+
+    bool add_check_packet(const uint8_t* buffer, uint32_t size, uint32_t dedup_period_ms)
+    {
+        using namespace std::chrono;
+        time_t timestamp = duration_cast<milliseconds>(std::chrono::system_clock::now() - _start_time).count();
+        // pop data from front queue, delete corresponding data from multiset
+        while (_time_hash_queue.size() > 0 && _time_hash_queue.front().first > timestamp + dedup_period_ms) {
```

Comment:
Shouldn't it be `stored_timestamp < (now - dedup_period)` because we want to clear all data older than $dedup_period seconds ago?

Reply:
After a second thought, `(timestamp - _time_hash_queue.front().first) > dedup_period_ms` might be easier to understand.

Reply:
Good catch! I prefer not to do subtractions with time to avoid getting negative numbers, so :
```suggestion
        while (_time_hash_queue.size() > 0 && timestamp > _time_hash_queue.front().first + dedup_period_ms) {
```